src/wp-includes/rest-api/endpoints/class-wp-rest-attachments-controller.php

```diff
@@ -135,6 +135,32 @@ public function create_item_permissions_check( $request ) {
 			);
 		}
 
+		/**
+		 * Filter whether the server should prevent uploads for image types it doesn't support. Default true.
+		 *
+		 * Developers can use this filter to enable uploads of certain image types.
+		 *
+		 * @param bool $check_mime Whether to prevent uploads of unsupported image types.
+		 */
+		$prevent_unsupported_uploads = apply_filters( 'wp_prevent_unsupported_image_uploads', true );
```

Comment:
I think we should pass the current mime type as a second argument. That gives developers more flexibility.

Reply:
Another thought: if in the long term we want to do support this for other mime types too, we can ditch the `_image_` from the filter name or use `_mime_type_`, e.g. `wp_prevent_unsupported_mime_type`. WDYT?

Reply:
👍🏼 `wp_prevent_unsupported_mime_type` -  I like it, makes sense to pass the mime type so developers can act on that, and the name should reflect that.

Reply:
Adding this here, but it will be null in `media_upload_form` because we don't know the mime type yet when the filter is run.

I wound up using `wp_prevent_unsupported_mime_type_uploads` as the name since this specifically applies to preventing uploads and this helps clarify that.